File: Sakila_Net/VB_SakilaLib/DAO/DAOBD/DAOBDPayment.vb
'using System;
Imports  System.Data
'using System.Collections.Generic;
'using System.Configuration;
'using System.Collections;
Imports SakilaLib.Cl.Harrys.Sakila.Model
Imports SakilaLib.Cl.Harrys.BD


Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace DAOBD

          Public class DAOPayment
             Public  Shared Function SelectAll(nPagina as Integer ,cantidad as Integer) as DataTable
                Dim StSql as String = "Select * from payment"
                Return MyBD.getInstance.SqlSelect(StSql, nPagina, cantidad)
        	   End Function

                    Public  Shared Function SelectOne( payment as ClPayment) as Boolean
             Dim StSql as String = "Select * from payment"
            StSql &= " Where  payment_id='" & payment.dataPaymentId & "'" _
                        & ""
                        Dim dt as DataTable = MyBD.getInstance.sqlSelect(stSql)
                        If dt.Rows.Count = 0 Then
                            Return False
                        End If
        payment.DataCustomerId = new ClCustomer(Integer.Parse(dt.Rows(0)("customer_id")))
        payment.DataStaffId = new ClStaff(Integer.Parse(dt.Rows(0)("staff_id")))
        payment.DataRentalId = new ClRental(Integer.Parse(dt.Rows(0)("rental_id")))
        payment.DataAmount = dt.Rows(0)("amount")
        payment.DataPaymentDate = dt.Rows(0)("payment_date").ToString()
        payment.DataLastUpdate = dt.Rows(0)("last_update").ToString()
                        Return True
                     end Function

        Public  Shared Function SqlInsert( payment as ClPayment)  as Integer
            Dim stSql as String = "insert into Payment (" _
                       & " payment_id ,customer_id ,staff_id ,rental_id ,amount ,payment_date ,last_update" _
                           & " )values (" _
            & "'" & payment.dataPaymentId & "'" _
                     & ",'" & payment.dataCustomerId.datacustomerId & "'" _
                     & ",'" & payment.dataStaffId.datastaffId & "'" _
                     & ",'" & payment.dataRentalId.datarentalId & "'" _
                     & ",'" & payment.dataAmount & "'" _
                     & ",'" & payment.dataPaymentDate & "'" _
                     & ",'" & payment.dataLastUpdate & "'" _
             & ")"
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

        Public  Shared Function SqlDelete( payment as ClPayment) as Integer
            
            Dim stSql as String =  "delete from Payment " _
             & " Where  payment_id='" & payment.dataPaymentId & "'" _
                       
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

       Public   Shared Function SqlUpdate(payment as ClPayment) as Integer
            Dim stSql as String =  "update Payment SET " _
             & " customer_id = '" & payment.DataCustomerId.DataCustomerId & "'" _
                             & ",staff_id = '" & payment.DataStaffId.DataStaffId & "'" _
                             & ",rental_id = '" & payment.DataRentalId.DataRentalId & "'" _
                             & ",amount = '" & payment.DataAmount & "'" _
                             & ",payment_date = '" & payment.DataPaymentDate & "'" _
                             & ",last_update = '" & payment.DataLastUpdate & "'" _
             & " Where  payment_id='" & payment.dataPaymentId & "'" _
& ""
            Return MyBD.getInstance.sqlEjecutar(stSql)
            
        end Function
          End Class

         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  

File: Sakila_Net/Sakila_Net/CS_SakilaLib/DAO/DAOBD/DAOBDPayment.cs
using System;
using  System.Data;
using System.Collections.Generic;
using System.Configuration;
using System.Collections;
using Cl.Harrys.Sakila.Model;
using Cl.Harrys.MyBD;


namespace Cl.Harrys.Sakila.DAOBD 
{ 
    public class DAOPayment 
    {
        public  static DataTable selectAll(int nPagina  ,int cantidad ) 
        {
            String StSql = "Select  payment_id ,customer_id ,staff_id ,rental_id ,amount ,payment_date ,last_update  from payment";
            return MyBD.MyBD.getInstance().sqlSelect(StSql, nPagina, cantidad);
        }

        public  static Boolean SelectOne( ClPayment payment)  
        {
            String stSql = "Select  payment_id ,customer_id ,staff_id ,rental_id ,amount ,payment_date ,last_update  from payment";
            stSql += " Where  payment_id='" + payment.PaymentId + "'" 
                        + "";
            DataTable dt = MyBD.MyBD.getInstance().sqlSelect(stSql,1,1);
            if (dt.Rows.Count == 0) 
                return false;
                        
        payment.CustomerId = new ClCustomer(int.Parse(dt.Rows[0]["customer_id"].ToString()));
        payment.StaffId = new ClStaff(int.Parse(dt.Rows[0]["staff_id"].ToString()));
        payment.RentalId = new ClRental(int.Parse(dt.Rows[0]["rental_id"].ToString()));
        payment.Amount = int.Parse(dt.Rows[0]["amount"].ToString());
        payment.PaymentDate = DateTime.ParseExact(dt.Rows[0]["payment_date"].ToString(), "yyyy-MM-dd HH:mm:ss,fff",System.Globalization.CultureInfo.InvariantCulture);
        payment.LastUpdate = DateTime.ParseExact(dt.Rows[0]["last_update"].ToString(), "yyyy-MM-dd HH:mm:ss,fff",System.Globalization.CultureInfo.InvariantCulture);
            return true;
        }

        public  static int sqlInsert( ClPayment payment)  
        {
            String stSql = "insert into Payment (" 
                       + " payment_id ,customer_id ,staff_id ,rental_id ,amount ,payment_date ,last_update" 
                           + " )values (" 
            + "'" + payment.PaymentId + "'" 
                     + ",'" + payment.CustomerId.CustomerId + "'" 
                     + ",'" + payment.StaffId.StaffId + "'" 
                     + ",'" + payment.RentalId.RentalId + "'" 
                     + ",'" + payment.Amount + "'" 
                     + ",'" + payment.PaymentDate + "'" 
                     + ",'" + payment.LastUpdate + "'" 
             + ")";
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

        public  static int sqlDelete( ClPayment payment) {
            
            String stSql  =  "delete from Payment " 
             + " Where  payment_id='" + payment.PaymentId + "'" 
                       ;
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

       public   static int sqlUpdate(ClPayment payment ) {
            String stSql  =  "update Payment SET " 
             + " customer_id = '" + payment.CustomerId.CustomerId + "'" 
                             + ",staff_id = '" + payment.StaffId.StaffId + "'" 
                             + ",rental_id = '" + payment.RentalId.RentalId + "'" 
                             + ",amount = '" + payment.Amount + "'" 
                             + ",payment_date = '" + payment.PaymentDate + "'" 
                             + ",last_update = '" + payment.LastUpdate + "'" 
             + " Where  payment_id='" + payment.PaymentId + "'" 
                 + "";
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
            
        }
   }

}
